### PR TITLE
chore: publish docs `tracing` for master + PR branches

### LINKED ISSUE #210: meta: CI publishes docs on master builds to GitHub Pages
### Motivation

It would be nice to auto-publish the master docs every time a commit is merged to master. This is especially helpful as there are a number of incubating nursery crates which have not been published to crates.io, and thus don't get published to docs.rs, but still see some use.

### Proposal

I believe https://github.com/crate-ci/azure-pipelines/issues/35 is tracking an upstream change to the CI configs that should hopefully make this easy. When that lands, we should add it to our build.

### PR DESCRIPTION
**Pull request by davidbarsky**
chore: build and publish documentation for each branch.

This commit introduces support for:
- building and deploying documentation for the master branch
- building and deploying (preview) documentation for each pull request and for each PR
- adds a link in `README.md` pointing to the generated documentation for the master branch
- notes the per-PR documentation previews in `CONTRIBUTING.md`

Closes #210

### CODE REVIEW

**Comment by hawkw** [comment]
I believe this closes #210 — can we add that to the PR description?

**Comment by hawkw** [comment]
The GitHub Actions breakage is unrelated and can be ignored; it's not blocking merges. I'll fix that.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -205,6 +205,8 @@ example would explicitly use `Timeout::new`. For example:
 /// # }
 ```
 
+To reduce the effort required to review documentation-related changes, `tracing`'s CI system generates preview websites containing the `rustdoc` output for the entire repository for each PR.
```
Comment (hawkw) [inline CONTRIBUTING.md]: Nit: can this be wrapped to match the rest of this file?

Comment (hawkw) [reply CONTRIBUTING.md]: Can we add instructions on how to actually view the docs previews here? Like:
> The preview can be viewed by clicking the `details` link on the `netlify/tracing-rs/deploy-preview` check on pull requests.

File: README.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ Application-level tracing for Rust.
 [gitter-url]: https://gitter.im/tokio-rs/tracing
```
Comment (hawkw) [inline README.md]: Can we also add `https://img.shields.io/badge/docs-master-blue` with a link to the master docs, preferably after the docs.rs badge?

File: CONTRIBUTING.md (hunk 1 above)
Comment (davidbarsky) [reply CONTRIBUTING.md]: fixed.

File: README.md (hunk 2 above)
Comment (davidbarsky) [reply README.md]: Done, added to all crates with a README.md.

Comment (davidbarsky) [reply README.md]: Done, added to all crates with a README.md.

**Pull request merged**